sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/grouping.scala

```diff
@@ -231,3 +229,30 @@ object GroupingID {
     if (SQLConf.get.integerGroupingIdEnabled) IntegerType else LongType
   }
 }
+
+object GroupingAnalytics {
+  def unapply(exprs: Seq[Expression])
```

Comment:
the `unapply` here should return None if the give expressions are not resolved yet.

Reply:
then we can fix https://github.com/apache/spark/pull/30144/files#r609770387

Reply:
> the `unapply` here should return None if the given expressions are not resolved yet.

Will throw other exception like before this pr

Reply:
What's wrong with my proposal that returning None if the given expressions are not resolved yet?
```
if (exprs.exists(!_.resolved)) {
  None
} else ...
```

Reply:
> What's wrong with my proposal that returning None if the given expressions are not resolved yet?
> 
> ```
> if (exprs.exists(!_.resolved)) {
>   None
> } else if (!exprs.exists(e => e.find(_.isInstanceOf[BaseGroupingSets]).isDefined)) ...
> ```

If here we return none, then it can't be resolved in current `ResolveGroupingAnalytics` then  will failed in `CheckAnalysis` when call datatype of CUBE/ROLLUP/GROUPING SETS and throw `UnsupportedOperationException`

Reply:
How about current